### PR TITLE
docs(appium): Add contributing documentation

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

Hola 👋 

It seems that the current contributing docs are empty. I took a stab at this to document some of the common workflows for folks to get started.

In full transparency: I added an option for users to check out the project with [Runme](https://runme.dev) which is a project I am working on that helps developers to get onboarded easier. Clicking on the link will just open VS Code, install the extension and open the same markdown document in a notebook renderer. This can be used as dashboard for further workflows, e.g. running tests or deploying docs. This is how it would look like:

appium

> Note: the `cd ../../../../../` are need as Runme uses the location of the markdown file as cwd. This will be updated [soon](https://github.com/stateful/runme/issues/263).

Please let me know what you think. I am actually doing the same in WebdriverIO where I will move all workflow scripts into markdown files and use the tool to trigger all workflows , e.g. also in CI. See [this pull request](https://github.com/webdriverio/webdriverio/pull/10237/files) as example. Again, this is totally optional to Appium devs.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

When checking out the project I failed to build the project due to:

```
packages/plugin-test-support/lib/harness.js:117:9 - error TS2322: Type 'string | undefined' is not assignable to type '(AddressConfig1 & string) | undefined'.
  Type 'string' is not assignable to type 'AddressConfig1 & string'.
    Type 'string' is not assignable to type 'AddressConfig1'.

117         address: host,
            ~~~~~~~

  node_modules/type-fest/source/simplify.d.ts:58:27
    58 export type Simplify<T> = {[KeyType in keyof T]: T[KeyType]} & {};
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The expected type comes from property 'address' which is declared here on type '{ subcommand?: CliCommand | undefined; configFile?: string | undefined; showConfig?: boolean | undefined; shell?: boolean | undefined; throwInsteadOfExit?: boolean | undefined; ... 32 more ...; tmpDir?: string | undefined; }'
```

Not sure if this is already a known issue. I added an ignore in my local version for the demo to work.

### CODE REVIEW

**Comment by jlipps** [comment]
@christian-bromann the type error is fixed by https://github.com/appium/appium/pull/18690 I think

File: packages/appium/docs/en/contributing/index.md [hunk 1]
```diff
@@ -16,8 +16,97 @@ already.
...
+
+To test your changes you can run Appium in dev mode via:
+
+```sh { name=start }
+npm start
+```
+
+### Run Tests
+
+The project maintains a set of different test variations you can run to verify the quality of the code.
```
Comment (jlipps) [inline packages/appium/docs/en/contributing/index.md:69]: can we add a note about running tests for a specific package? e.g. `cd packages/$package` and `npm run test`

**Review by jlipps — CHANGES_REQUESTED** [review]

File: packages/appium/docs/en/contributing/index.md (hunk 1 above)
Comment (christian-bromann) [reply packages/appium/docs/en/contributing/index.md:70]: add example to trigger test from a different workspace:
````
You can also run tests for specific workspaces, e.g.:

```sh { name=test-workspace }
export APPIUM_WORKSPACE=@appium/doctor
npm run test:unit -w $APPIUM_WORKSPACE
```
````

**Comment by mykola-mokhnach** [comment]
Hello @christian-bromann 

Thanks for taking care about that

File: packages/appium/docs/en/contributing/index.md [hunk 2]
```diff
@@ -16,8 +17,104 @@ already.
...
 
-TODO
+The project offers a variety of ways to contribute, e.g.:
+
+- contributing code
+- improving documentation
+- creating educational content (blog posts, tutorials, videos, etc.)
+- spreading the good word about the project (e.g. via Twitter)
+- filing bugs if you discover them while using Appium
+- making feature requests if you are missing something in the project
```
Comment (mykola-mokhnach) [inline packages/appium/docs/en/contributing/index.md:27]: triaging bugs and helping others in [Appium forum](https://discuss.appium.io) is also highly appreciated

Comment (christian-bromann) [reply packages/appium/docs/en/contributing/index.md]: added 👍

**Review by jlipps — APPROVED** [review]
LGTM

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by christian-bromann** [comment]
Thanks for merging @KazuCocoa , any ETA when this will be deployed to appium.io?

**Comment by KazuCocoa** [comment]
i don't know current steps to publish a new doc in this repo.
cc @jlipps

**Pull request merged**